src/analysis.py

```diff
@@ -0,0 +1,219 @@
+    train_test_split,
+)
+from sklearn.pipeline import Pipeline, make_pipeline
+from sklearn.preprocessing import OneHotEncoder, OrdinalEncoder, StandardScaler
+from sklearn.svm import SVC, SVR
+
+import altair as alt
+from vega_datasets import data
+
+from docopt import docopt
+from pylab import savefig
+from render_table import render_table
+
+print("start")
+args = docopt(__doc__)
+
+def read_input_file(input_file_path):
+    """
+    Validates input file path.
+    Parameters:
+    -----------
+    input_file_path : str
+        input path to be verified
+
+    Returns:
+    -----------
+    pandas.DataFrame
+        if path is valid and verified
+    """
+    if not os.path.isfile(input_file_path):
+        print("Input file does not exist.")
+        sys.exit()
+
+    try:
+        data_frame = pd.read_csv(input_file_path, index_col=0, parse_dates = ['first_appearance'])
+        print('Path is valid')
+    except:
+        print(input_file_path + 'Path is not valid. Please check ')
+        sys.exit()
+
+    combined_columns = ['name', 'id', 'align', 'eye', 'hair', 'sex', 'gsm','appearances', 'first_appearance', 'year', 'publisher']
+
+    if not all([item in data_frame.columns for item in combined_columns]):
+        print(input_file_path + " should contain these columns: " + str(combined_columns))
+        sys.exit()
+
+    print('Returning data frame')
+    return data_frame
+
+#This function is adapted from Varada's lecture code in DSCI571
+def store_cross_val_results(model, scores, results):
+    """
+    Stores mean scores from cross_validate in results_dict for
+    the given model model_name.
+
+    Parameters
+    ----------
+    model :
+        scikit-learn classification model
+    scores : dict
+        object return by `cross_validate`
+    results_dict: dict
+        dictionary to store results
+
+    Returns
+    ----------
+        None
+
+    """
+    results[model] = {
+        "mean_validation_accuracy": "{:0.4f}".format(np.mean(scores["test_score"])),
+        "mean_train_accuracy": "{:0.4f}".format(np.mean(scores["train_score"])),
+        "mean_fit_time (s)": "{:0.4f}".format(np.mean(scores["fit_time"])),
+        "mean_score_time (s)": "{:0.4f}".format(np.mean(scores["score_time"])),
+        "std_train_score": "{:0.4f}".format(scores["train_score"].std()),
+        "std_test_score": "{:0.4f}".format(scores["test_score"].std()),
+    }
+
+
+
+def train_model(dataframe):
+    """
+    Processes the data, trains the model, and returns a dataframe showing the results
+
+    Parameters
+    ----------
+    dataframe : pd.Dataframe
+        Dataframe to be used in model
+
+    Returns
+    ----------
+        pandas.Dataframe
+
+    """
+    train_df = dataframe
+    X_train = train_df.drop(columns=['align'])
+    y_train = train_df['align']
+
+    numeric_features = ['appearances']
+    categorical_features = ['name', 'id', 'eye', 'hair', 'year', 'publisher', 'sex']
+    drop_features = ['gsm', 'first_appearance']
```

Comment:
Agreed on dropping gsm, it's too sparse. First appearance we can probably also leave out of our initial analysis. We would probably want to trim it to just the month to avoid duplicating year data as part of feature engineering.